pkg/util/retryutil/retry_util.go

```diff
@@ -5,14 +5,39 @@ import (
 	"time"
 )
 
-// Retry calls ConditionFunc until it returns boolean true, a timeout expires or an error occurs.
+type RetryTicker interface {
+	Stop()
+	Tick()
+}
+
+type Ticker struct {
+	ticker *time.Ticker
+}
+
+func (t *Ticker) Stop() { t.ticker.Stop() }
+
+func (t *Ticker) Tick() { <-t.ticker.C }
+
+// Retry calls ConditionFunc until either:
+// * it returns boolean true
+// * a timeout expires
+// * an error occurs
 func Retry(interval time.Duration, timeout time.Duration, f func() (bool, error)) error {
 	//TODO: make the retry exponential
 	if timeout < interval {
 		return fmt.Errorf("timout(%s) should be greater than interval(%v)", timeout, interval)
 	}
+	tick := &Ticker{time.NewTicker(interval)}
+	return RetryWorker(interval, timeout, f, tick)
+}
+
+func RetryWorker(
+	interval time.Duration,
+	timeout time.Duration,
+	f func() (bool, error),
+	tick RetryTicker) error {
```

Comment:
I don't think this is too important, but maybe the function parameter should be at the end to make invocations with long hanging lambdas more readable.

Reply:
Yeah, it actually bothered me too a bit.